src/shim.h

```diff
@@ -18,44 +18,70 @@
  * This would be signal fd, stdin fd, proxy socket fd and an I/O
  * fd passed by the runtime
  */
-#define MAX_POLL_FDS 4
+#define MAX_POLL_FDS 3
 
 struct cc_shim {
 	char       *container_id;
 	int         proxy_sock_fd;
-	int         proxy_io_fd;
-	uint64_t    io_seq_no;
-	uint64_t    err_seq_no;
-	bool        exiting;
+	char       *token;
+	char       *proxy_address;
+	int         proxy_port;
 };
 
-/*
- * control message format
- * | ctrl id | length  | payload (length-8)      |
- * | . . . . | . . . . | . . . . . . . . . . . . |
- * 0         4         8                         length
- */
-#define CONTROL_HEADER_SIZE             8
-#define CONTROL_HEADER_LENGTH_OFFSET    4
-
-/*
- * stream message format
- * | stream sequence | length  | payload (length-12)     |
- * | . . . . . . . . | . . . . | . . . . . . . . . . . . |
- * 0                 8         12                        length
- */
-#define STREAM_HEADER_SIZE              12
-#define STREAM_HEADER_LENGTH_OFFSET     8
-
-#define PROXY_CTL_HEADER_SIZE           8
-#define PROXY_CTL_HEADER_LENGTH_OFFSET  0
-
-/*
- * Hyperstart is limited to sending this number of bytes to
- * a client.
- *
- * (This value can be determined by inspecting the hyperstart
- * source where hyper_event_ops->wbuf_size is set).
- */
-#define HYPERSTART_MAX_RECV_BYTES       10240
+// Header size is length of header in 32 bit words.
+#define  MIN_HEADER_SIZE    3
```

Comment:
All other sizes are in bytes, so this could be the source of possible errors as the assumption will be that's 3 bytes. Can it be expressed in bytes for consistency instead or renamed to make it clear this is *not* in bytes?